File: sample_vuln/insecure.py
# Amostra PROPOSITALMENTE vulnerável — usada só pelo smoke test do CI.
# Não é código real; serve para provar que o raptor-win detecta algo.
import hashlib
import subprocess


def weak_hash(data: bytes) -> str:
    return hashlib.md5(data).hexdigest()  # weak hash (esperado)


def run(user_input: str) -> None:
    subprocess.run(user_input, shell=True)  # command injection (esperado)

# XPath injection: input do usuario concatenado na consulta (esperado)
def buscar(tree, request):
    termo = request.GET.get("q")
    return tree.xpath("//user[name='" + termo + "']")

# Prompt injection: input do usuario vai direto ao prompt do LLM (esperado)
def resumir(client, request):
    texto = request.POST.get("texto")
    return client.chat.completions.create(model="gpt-x", messages=[{"role": "user", "content": texto}])

File: sample_vuln/insecure.js
// Amostra PROPOSITALMENTE vulnerável — usada só pelo smoke test do CI.
const express = require('express')
const app = express()

app.get('/proxy', (req, res) => {
  // SSRF: URL controlada pelo cliente vai direto para um fetch (esperado)
  fetch(req.query.url).then(r => r.text()).then(t => res.send(t))
})

// SSRF de Web Push: o endpoint da inscrição vem do navegador do assinante e vai
// direto para o fetch, sem allowlist de host (esperado por raptor.ssrf.web-push-endpoint)
async function enviarPush(sub) {
  await fetch(sub.endpoint, { method: 'POST', headers: { TTL: '60' } })
}

module.exports = { app, enviarPush }
